docker/docker_client.go

```diff
@@ -262,26 +272,31 @@ func (c *dockerClient) setupRequestAuth(req *http.Request) error {
 		req.SetBasicAuth(c.username, c.password)
 		return nil
 	case "bearer":
-		realm, ok := challenge.Parameters["realm"]
-		if !ok {
-			return errors.Errorf("missing realm in bearer auth challenge")
-		}
-		service, _ := challenge.Parameters["service"] // Will be "" if not present
-		scope := fmt.Sprintf("repository:%s:%s", c.scope.remoteName, c.scope.actions)
-		token, err := c.getBearerToken(realm, service, scope)
-		if err != nil {
-			return err
+		now := time.Now()
+		if now.After(c.tokenExpiration) || c.token == nil {
```

Comment:
Switch the order of the checks; `c.tokenExpiration` is uninitialized if `c.token == nil` (and the `c.token == nil` check is more efficient, but that should not be the decisive criterion).

Is the `now` variable necessary? AFAICS it has a single user.